libraries/provider_install_source.rb

```diff
@@ -0,0 +1,68 @@
+class Chef
+  # Chef Provider for installing or removing Elasticsearch from package or source
+  class Provider::ElasticsearchInstallSource < Provider
+    include Poise
+    include ElasticsearchCookbook::Helpers
+    if respond_to?(:provides)
+      provides :elasticsearch_install
+    end
+
+    def action_install
+      converge_by("#{new_resource.name} - install source") do
+        notifying_block do
+          include_recipe 'ark'
```

Comment:
Argh :)) Are we really sure we want to rely on the `ark` again? I can see there's some activity in the new repository, but I'd like us to move away from it, and just do all the extract/move/symlink acrobatics ourselves, even though it's more work...